airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/concurrent/adapters.py

```diff
@@ -243,6 +243,9 @@ def read(self) -> Iterable[Record]:
                     data_to_return = dict(record_data)
                     self._stream.transformer.transform(data_to_return, self._stream.get_json_schema())
                     yield Record(data_to_return, self.stream_name())
+                elif isinstance(record_data, AirbyteMessage) and hasattr(record_data, "record"):
```

Comment:
nit: can we check the record type instead of using hasattr? `message.getType() == AirbyteMessage.Type.RECORD`

Reply:
Ah, I'd forgotten about that. Done!